components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/RegistryRecoveryDataStore.java

```diff
@@ -276,25 +276,39 @@ private void deleteOldResourcesIfFound(Registry registry, String userName, Strin
             }
         }
 
+        boolean isTransactionSucceeded = false;
         try {
             if (collection != null) {
                 String[] resources = collection.getChildren();
                 for (String resource : resources) {
+                    isTransactionSucceeded = false;
                     String[] splittedResource = resource.split("___");
                     if (splittedResource.length == 3) {
                         //PRIMARY USER STORE
                         if (resource.contains("___" + userNameToValidate.toLowerCase() + "___")) {
-
-                            registry.beginTransaction();
-                            // Check whether the resource still exists for concurrent cases.
-                            if (registry.resourceExists(resource)) {
-                                registry.delete(resource);
-                                registry.commitTransaction();
-                            } else {
-                                // Already deleted by another thread. Do nothing.
-                                registry.rollbackTransaction();
-                                if (log.isDebugEnabled()) {
-                                    log.debug("Confirmation code already deleted in path of resource : " + resource);
+                            try {
+                                registry.beginTransaction();
+                                // Check whether the resource still exists for concurrent cases.
+                                if (registry.resourceExists(resource)) {
+                                    registry.delete(resource);
+                                    isTransactionSucceeded = true;
+                                } else {
+                                    // Already deleted by another thread. Do nothing.
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Confirmation code already deleted in path of resource : " + resource);
+                                    }
+                                }
+                            } catch (RegistryException ex) {
+                                log.error("Error while deleting the old confirmation code \n" + ex);
```

Comment:
The Throwable class has overridden the toString() method.
So, printing the exception does not print it as an object.
Therefore, I think we can keep the error log as it is.
Additionally will add a debug log to print the stacktrace.
Since this is in a for loop, printing the error stacktrace can consume the log file space.